pyvista/core/filters/image_data.py

```diff
@@ -1604,3 +1606,287 @@ def _update_and_get_output():
         # Restore active scalars
         self.set_active_scalars(scalars, preference='point')  # type: ignore[attr-defined]
         return output
+
+    def label_connectivity(
+        self,
+        *,
+        scalars: str | None = None,
+        scalar_range: str | VectorLike[int] | VectorLike[float] | None = 'auto',
```

Comment:
With type hints, `float` also implies `int`, and so specifying both is redundant. See 
https://peps.python.org/pep-0484/#the-numeric-tower

Also suggest using Literal if `auto` is the only acceptable string
```suggestion
        scalar_range: Literal['auto'] | VectorLike[float] | None = 'auto',
```